--- hwif/hal-lzpi.c	2019-05-30 08:48:04.405982179 +0900
+++ hwif/hal-lzpi.c_20190529	2019-05-30 08:48:04.405982179 +0900
@@ -129,7 +129,7 @@
 void HAL_set_timer0_function(void (*func)(uint32_t sys_timer_count)) {
 	syslog_timer_ext_func = func;
 }
-static void syslog_timer_isr(struct timer_list *t) {
+static void syslog_timer_isr(unsigned long data) {
 	if(syslog_timer_ext_func) syslog_timer_ext_func(jiffies);
 	syslog_timer.expires = jiffies + HZ*2;
 	add_timer(&syslog_timer);
@@ -138,8 +138,9 @@
 {
 	// system logging start
 	syslog_timer_ext_func = NULL;
-	timer_setup(&syslog_timer,syslog_timer_isr,0);
+	init_timer(&syslog_timer);
 	syslog_timer.expires = jiffies + HZ*2;
+	syslog_timer.function = syslog_timer_isr;
 	add_timer(&syslog_timer);
 
 	m.trigger=0;
@@ -484,7 +485,7 @@
 }
 
 static bool timer_flag=false;
-void timer_function(struct timer_list *t)
+void timer_function(unsigned long data)
 {
 	//printk(KERN_INFO"%s %s %d\n",__FILE__,__func__,__LINE__);
 	if(ext_timer_func){
@@ -504,9 +505,11 @@
 	uint32_t add_time;
 	add_time = (tmp*msec)/1000; 
 	if(add_time<2) add_time = 2;
-	timer_setup(&g_timer,timer_function,0);
+	init_timer(&g_timer);
+	g_timer.data = 0;
 	g_timer.expires = jiffies + add_time;
 	//	printk(KERN_INFO"%s %s %d   addtime=%d\n",__FILE__,__func__,__LINE__,add_time);
+	g_timer.function = timer_function;
 	ext_timer_func = func;
 	add_timer(&g_timer);
 	timer_flag = true;
